### PR TITLE
Inheritance

### PR DESCRIPTION
**Pull request by cerilacanda**

### CODE REVIEW

File: content/Inheritance/Inheritance.facts.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+The `protected` visibility keyword allows a child class to access the properties and methods of a parent class.
+
+A child class can override the methods of its parent class.
+
+Defining a method in a child class with the same name and parameter of a parent class method is called `overriding`.
+
+A process of changing the behavior of a parent class method in a child class is called `method overriding`.
+
+The `parent` keyword with the scope resolution operator `::` is used to access a parent class methods in a child class.
```
Comment (redgeneLauren) [inline content/Inheritance/Inheritance.facts.md]: Kindly check this out. Thanks.
InheritanceFactsGrammarly

File: content/Inheritance/Inheritance.facts.md [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+```
+
+Output:
+```
+Diana is 12 years old.Anna is taking up BSIT.
+```
+
+In the `TestStudent` class declaration above, the code breaks down as follows:
+
+ - `class TestStudent extends TestPerson` declares the `TestStudent` class as the child class of the `TestPerson` class
```
Comment (redgeneLauren) [inline content/Inheritance/Inheritance.facts.md]: Lacks period `.` at the end:
.... of the `TestPerson` class.

File: content/Inheritance/Inheritance.facts.md [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+ - `private $course;` is the definition of the `$course` property without a default value in the `TestStudent` class.
+
+ - `public function __construct($name, $age, $course) { }` is the constructor definition of the `TestStudent` class with the parameters `$name`, `$age`, and `$course`. It is automatically called when a new object is created.
+
+ - `parent::__construct($name, $age);` calls the `__construct()` method of the parent class `TestPerson` passing the arguments `$name` and `$age`.
+
+ - `$this->course = $course;` assigns the value of the variable `$course` to the `$course` property of the `TestStudent` class.
+
+ - `public function display() {...}` the method definition that overrides the `display()` method of the parent class `TestPerson`.
```
Comment (redgeneLauren) [inline content/Inheritance/Inheritance.facts.md]: Do you mean:
`public function display() {...} is the method definition that overrides...`
Or
`The method definition public function display() {...} overrides the ....`

File: content/Inheritance/Inheritance.q.md [hunk 4]
```diff
@@ -0,0 +1,2538 @@
...
+    }
+}
+
+$studObject = new Student("John", 20, "BSCS");
+$studObject->display();
+?>
+```
+/// type=SS, answer=[5]
+
+In the `echo` statement on line 19 of `Student.php`, replace `parent::getName()` with `$this->name`. Execute the program? What is its output?
```
Comment (redgeneLauren) [inline content/Inheritance/Inheritance.q.md]: Typo: `Execute the program.`

File: content/Inheritance/Inheritance.q.md [hunk 5]
```diff
@@ -0,0 +1,2538 @@
...
+    }
+}
+
+$studObject = new Student("John" 20, "BSCS");
+studObject->display();
+?>
+```
+/// type=SS, answer=[5]
+
+remove the `private` visibility keyword in the property definition `private $course;` on line 4 of `Student.php`. Execute the program. What is the error message?
```
Comment (redgeneLauren) [inline content/Inheritance/Inheritance.q.md]: Typo: `Remove`

**Review by redgeneLauren — APPROVED** [review]

**Review by rosoarv — APPROVED** [review]

**Review by gjoren4 — APPROVED** [review]

**Pull request merged**